### DISCUSSION TITLE
Date Added - Sorting or Batch Tagging

### DISCUSSION THREAD

**Discussion by almant** [Difegue/LANraragi#479] [Q&A]
Currently LANraragi needs a lot of input to display most recent galleries.
This is partially because it doesn't save the sorting setting, but also because so many of my archives were imported before date_added was a thing.
As a result if I click date_added (descending) I'll get the oldest archives from after the namespace was introduced (not the actual oldest). Then if I click it again (ascending) I won't get the newest archives because the archives without the namespace (i.e the oldest archives in my collection) are ordered before the actual newest archives.

So currently I need to search for date_added to filter out archives without a date_added value, then click date_added twice to actually get the most recent archives.

- A super basic solution for this would be to allow batch tagging of every archive, to have date_added set to 1 if they don't have that data already. This would at least order the oldest archives in a logical way and reduce the number of steps to see your most recent archives.

- A more complex but less hacky solution would be to have date_added ordering omit archives without a value for that namespace (or at least sort them in a more logical way).

- And probably the most ideal but harder to implement feature would be maintaining default views (either through cookies or server-side). I'd really prefer to see my most recently imported archives as soon as I open the LANraragi webpage and I'm sure I'm not the only one.

Anyway, thanks for all the work on the project. LANraragi really is the best reading server right now and I've loved all the updates. Keep up the amazing work!

**Comment by Difegue** [comment]
Thanks to #336 you should be able to carry your entire search workflow as URL parameters by the next release, so you won't have to do the search/sort dance every time at least. ✌

You could also transfer your search to a dynamic category if you want, but I don't think that'll be very necessary with the upcoming URL changes. 🤔

Past that, you could also add a bogus `date_added` tag to all the archives that don't have one using Batch Tagging and the Tag Copier plugin, but that does require you to check them manually in the Batch window, which might be a bit unwieldy if you have a lot.

**Comment by Difegue** [comment]
I'm moving this to a discussion as URL parameters for search have been released for a little while now, but I'll try to keep in mind the suggestions about default views and ordering.  

Thanks again for the input! 🙏
